chore: add manual backport instructions to readme (#594)
leanprover/reference-manual: README.md

Then open <http://localhost:8880> in your browser.

## Backporting Manual Changes

> In the event that a backported update is necessary, manual deployment of the newly built HTML is handled via Git tags, which are version-numbered tags based on major Lean versions matching the RegEx pattern `^v4\.\d+\.\d+$`. Pushing a new tag triggers the necessary deployment, but to do so requires first deleting any existing tag of the same name. The following describes the steps and commands you may use to manage this process.

Reviewer: ```suggestion
The manual for a Lean version is automatically deployed when a Git tag that matches the version is pushed. For Lean version `4.X.Y`, the manual is deployed from the contents of GIt tag `v4.X.Y` (note the leading `v`). In the event that a backported update is necessary, the tag should be changed. Pushing a new tag triggers the necessary deployment, but to do so requires first deleting any existing tag of the same name, because Git doesn't  support modification of existing tags. The following describes the steps and commands you may use to manage this process.
```

Make any desired changes to the codebase. Once you have tested changes locally, create a new branch based on the version of the reference manual you are modifying. By convention, this branch should match the tag (i.e. version number), but without the leading `v`. Push this branch to the remote.

> For example, for modifications to the reference manual associated with Lean version 4.22, the branch should be named `4.22.0`:

Reviewer: ```suggestion
For example, for modifications to the reference manual associated with Lean version 4.22, the branch should be named `4.22.0`. These examples assume that the main reference manual repository is named `origin` in your local checkout:
```

```

> git checkout -b 4.22.0  //create new branch

Reviewer: Here, we should add a step to check if the branch exists before just creating it. If there's already a maintenance branch for a given release, it should be used.

Reviewer: It won't let you push if you don't have all the changes in it. Maybe this should be phrased in Git verbs rather than commands?

git push origin 4.22.0  //push new branch to remote
```

Any existing tag matching the desired version number must be deleted on both the remote and locally, then regeneratged locally and pushed to the remote to trigger deployment. 

For example, for modifications to the reference manual associated with Lean version 4.22, the tag to be regenerated is `v4.22.0` and the commands to delete it are:

```

> git push origin :v4.22.0  //delete remote tag
> git tag --delete v4.22.0  //delete local tag

Reviewer: ```suggestion
git push origin :v4.22.0  # delete remote tag
git tag --delete v4.22.0  # delete local tag
```
